docs/proposals/scheduling/20221227-cpu-static-reservation.md

```diff
@@ -0,0 +1,176 @@
+---
+title: resource reservation
+authors:
+  - "@lucming"
+reviewers:
+  - "@eahydra"
+  - "@hormes"
+  - "@yihuifeng"
+  - "@honpey"
+  - "@zwzhang0107"
+  - "@jasonliu747"
+creation-date: 2022-12-27
+last-updated: 2022-12-27
+---
+
+# Resource Reservation
+
+## Table of Contents
+
+<!--ts-->
+
+- [Resource Reservation](#resource-reservation)
+  - [Table of Contents](#table-of-contents)
+  - [Summary](#summary)
+  - [Motivation](#motivation)
+    - [Goals](#goals)
+    - [Non-Goals/Future Work](#non-goalsfuture-work)
+  - [Proposal](#proposal)
+    - [User Stories](#user-stories)
+      - [Story 1](#story-1)
+      - [Story 2](#story-2)
+      - [Story 3](#story-3)
+    - [Implementation Details](#implementation-details)
+      - [koordlet](#koordlet)
+      - [koord-manager](#koord-manager)
+      - [koord-scheduler](#koord-scheduler)
+      - [example](#example)
+  - [Implementation History](#implementation-history)
+<!--te-->
+
+## Summary
+The proposal provides a scheduling mechanism to statically reserve resources on nodes, such as reserving 0-6 cores of cpu on the node.
+
+## Motivation
+
+
+### Goals
+
+- statically reserving cpu resources on node.
+- support for reserving a number of cpus on the node.
+- support for reserving a number of memory on the node.
+
+### Non-Goals/Future Work
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+As a k8s cluster administrator, there are not only containers running on the nodes of the k8s cluster, but there may be other processes that are not controlled by k8s but are just normal processes, 
+in order to ensure the isolation of the containers and the original processes. 
+At the cpu level, we usually want normal processes and containers to run on different cpu cores to reduce/avoid the impact of their resource usage.   
+eg：   
+totoal cpu on node: 24;   
+expected performance maybe like this:   
+normal process outside k8s use: 0-11;    
+container use: 12-23   
+so for k8s, the 0-11 core is reserved and it cannot be used.
+
+#### Story 2
+As a k8s cluster administrator, set the total amount of resources to be reserved directly via `node.annotation`, just like: `1C2Gi`.
+
+#### Story 3
+As a k8s cluster administrator, set the total number of cpu cores to be reserved via kubelet, and then set the specific cpu cores to be reserved via `node.annotation`, eg: kubelet sets "4C", node.annaion sets the cores with id 0-3 not to be used.
+
+### Implementation Details
+By adding annotation to the node, we specify the resources to be reserved by the koordinator, such as memory, cpu, etc. For cpu, we can specify the total amount of cpu to be reserved, or we can specify which cores to explicitly reserve. 
+```golang
+// resources that koordinator reserved, and you can reserve other resource if needed.
+type KoordReserved struct {
+	// total cpus need to be reserved, just like "2C"
+	CPUNums      string `json:"CPUNums"`
```

Comment:
Consider using corev1.ResourceList to describe the resources and capacity to be reserved